common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -153,37 +153,39 @@ public Object run() {
-                            // Will throw UnsupportedOperationException if unsafe memory access is denied:
-                            long address = finalUnsafe.allocateMemory(Long.BYTES);
-                            finalUnsafe.putLong(address, 42);
-                            finalUnsafe.freeMemory(address);
+                            if (javaVersion() > 23) {
+                                cls.getDeclaredMethod("objectFieldOffset", Field.class);
+                                cls.getDeclaredMethod("staticFieldOffset", Field.class);
+                                cls.getDeclaredMethod("staticFieldBase", Field.class);
+                                cls.getDeclaredMethod("arrayBaseOffset", Class.class);
+                                cls.getDeclaredMethod("arrayIndexScale", Class.class);
+                                cls.getDeclaredMethod("allocateMemory", long.class);
+                                cls.getDeclaredMethod("reallocateMemory", long.class, long.class);
+                                cls.getDeclaredMethod("freeMemory", long.class);
+                                cls.getDeclaredMethod("setMemory", long.class, long.class, byte.class);
+                                cls.getDeclaredMethod("setMemory", Object.class, long.class, long.class, byte.class);
+                                cls.getDeclaredMethod("getBoolean", Object.class, long.class);
+                                cls.getDeclaredMethod("getByte", long.class);
+                                cls.getDeclaredMethod("getByte", Object.class, long.class);
+                                cls.getDeclaredMethod("getInt", long.class);
+                                cls.getDeclaredMethod("getInt", Object.class, long.class);
+                                cls.getDeclaredMethod("getLong", long.class);
+                                cls.getDeclaredMethod("getLong", Object.class, long.class);
+                                cls.getDeclaredMethod("putByte", long.class, byte.class);
+                                cls.getDeclaredMethod("putByte", Object.class, long.class, byte.class);
+                                cls.getDeclaredMethod("putInt", long.class, int.class);
+                                cls.getDeclaredMethod("putInt", Object.class, long.class, int.class);
+                                cls.getDeclaredMethod("putLong", long.class, long.class);
+                                cls.getDeclaredMethod("putLong", Object.class, long.class, long.class);
+                                cls.getDeclaredMethod("addressSize");
+                                // The following tests the methods are usable.
+                                // Will throw UnsupportedOperationException if unsafe memory access is denied:
+                                long address = finalUnsafe.allocateMemory(8);
+                                finalUnsafe.putLong(address, 42);
+                                finalUnsafe.freeMemory(address);
```

Comment:
This section should be outside of the version check clause, as they can fail on JDK 23.

Reply:
Could you elaborate? My understanding of https://openjdk.org/jeps/471 is It ~could~ fail if `--sun-misc-unsafe-memory-access` set to bad value, but netty handles that in other conditional branch. On same 471 they say `Deprecate all of the memory-access methods — on-heap, off-heap, and bimodal — for removal`, so no removal will happen in jdk23. https://github.com/openjdk/jdk/blob/jdk23/src/java.base/share/classes/jdk/internal/misc/Unsafe.java#L3824 does not ring any bells either.

Reply:
> My understanding of https://openjdk.org/jeps/471 is It could fail if --sun-misc-unsafe-memory-access set to bad value, but netty handles that in other conditional branch

The system property we use is undocumented, so it could change or work differently across different patch releases or JDK distros. Hence I'd like some memory access calls as a fallback check.

The reflective checks are fine to be guarded by the version check, as you explain here an in the PR description.